### PR TITLE
Feature/163005226 no rabbit

### PR DESCRIPTION
**Pull request by rockem**

### CODE REVIEW

File: src/main/java/io/ticktok/server/clock/V1Controller.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
+package io.ticktok.server.clock;
+
+import org.springframework.core.annotation.AliasFor;
+import org.springframework.stereotype.Component;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.lang.annotation.*;
```
Comment (amaziagur) [inline src/main/java/io/ticktok/server/clock/V1Controller.java]: you should not use versioning in class name - `V1Controller`.
this is why we use versioning in path params (or header)

File: src/main/java/io/ticktok/server/tick/http/MongoHttpQueuesRepository.java [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    }
+
+    @PostConstruct
+    public void createTTLIndex() {
+        try {
+            mongo.indexOps(HttpQueue.class).dropIndex(QUEUE_TTL);
+        } catch (UncategorizedMongoDbException e) {
+            // Index not found, ignore
+        }
+        mongo.indexOps(HttpQueue.class).ensureIndex(
```
Comment (amaziagur) [inline src/main/java/io/ticktok/server/tick/http/MongoHttpQueuesRepository.java]: can you maybe extract it to method? for clean and readable

File: src/test/kotlin/e2e/test/io/ticktok/server/support/Client.kt [hunk 3]
```diff
@@ -35,72 +45,133 @@ object Client {
...
+            if (connection == null) {
+                connection = createConnection(clock.channel)
+                channel = connection!!.createChannel()
+            }
         }
 
-        fun stop() {
+        private fun createConnection(channel: ClockChannel?): Connection {
+            val factory = ConnectionFactory()
+            factory.setUri(channel?.details!!["uri"])
```
Comment (amaziagur) [inline src/test/kotlin/e2e/test/io/ticktok/server/support/Client.kt]: this is actually not safe code.
you should use `let{}`
https://discuss.kotlinlang.org/t/let-vs-if-not-null/3542/11

File: src/test/kotlin/e2e/test/io/ticktok/server/support/Client.kt [hunk 4]
```diff
@@ -35,72 +45,133 @@ object Client {
...
+                channel!!.basicCancel(consumerTag)
+            }
+        }
+    }
+
+    class HttpTickListener(clock: Clock) : TickListener(clock) {
+
+        var listenerTimer: Timer = Timer(true);
+
+        override fun listenOn(clock: Clock) {
```
Comment (amaziagur) [inline src/test/kotlin/e2e/test/io/ticktok/server/support/Client.kt]: you should have 1 lvl of indentation

File: src/main/java/io/ticktok/server/clock/V1Controller.java (hunk 1 above)
Comment (rockem) [reply src/main/java/io/ticktok/server/clock/V1Controller.java]: fixed

File: src/main/java/io/ticktok/server/tick/http/MongoHttpQueuesRepository.java (hunk 2 above)
Comment (rockem) [reply src/main/java/io/ticktok/server/tick/http/MongoHttpQueuesRepository.java]: fixed

File: src/test/kotlin/e2e/test/io/ticktok/server/support/Client.kt (hunk 3 above)
Comment (rockem) [reply src/test/kotlin/e2e/test/io/ticktok/server/support/Client.kt]: Can you give an example of how should I do it?

Comment (rockem) [reply src/test/kotlin/e2e/test/io/ticktok/server/support/Client.kt]: Can you give an example of how should I do it?

File: src/test/kotlin/e2e/test/io/ticktok/server/support/Client.kt (hunk 4 above)
Comment (rockem) [reply src/test/kotlin/e2e/test/io/ticktok/server/support/Client.kt]: fixed

File: src/test/kotlin/e2e/test/io/ticktok/server/support/Client.kt (hunk 3 above)
Comment (amaziagur) [reply src/test/kotlin/e2e/test/io/ticktok/server/support/Client.kt]: sure, please see this example:
To perform a certain operation only for non-null values, you can use the safe call operator together with let

```
val listWithNulls: List<String?> = listOf(“A”, null)
for (item in listWithNulls) {
 item?.let { println(it) } // prints A and ignores null
}
```

Comment (amaziagur) [reply src/test/kotlin/e2e/test/io/ticktok/server/support/Client.kt]: please see above

Comment (rockem) [reply src/test/kotlin/e2e/test/io/ticktok/server/support/Client.kt]: I don't understand how can I do it on my code, I want it to fail if there's no uri

**Pull request merged**